pkg/compute/deploy.go

```diff
@@ -41,137 +56,437 @@ func NewDeployCommand(parent common.Registerer, client api.HTTPClient, globals *
 	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
 	c.CmdClause.Flag("version", "Number of version to activate").Action(c.Version.Set).IntVar(&c.Version.Value)
 	c.CmdClause.Flag("path", "Path to package").Short('p').StringVar(&c.Path)
+	c.CmdClause.Flag("domain", "The name of the domain associated to the package").StringVar(&c.Domain)
+	c.CmdClause.Flag("backend", "A hostname, IPv4, or IPv6 address for the package backend").StringVar(&c.Backend)
+	c.CmdClause.Flag("backend-port", "A port number for the package backend").UintVar(&c.BackendPort)
 
 	return &c
 }
 
 // Exec implements the command interface.
 func (c *DeployCommand) Exec(in io.Reader, out io.Writer) (err error) {
-	var progress text.Progress
-	if c.Globals.Verbose() {
-		progress = text.NewVerboseProgress(out)
-	} else {
-		progress = text.NewQuietProgress(out)
+	// Exit early if no token configured.
+	_, s := c.Globals.Token()
+	if s == config.SourceUndefined {
+		return errors.ErrNoToken
 	}
 
+	var (
+		progress text.Progress
+		version  *fastly.Version
+		desc     string
+	)
+
+	progress = newProgress(c.Globals.Verbose(), out)
+
 	defer func() {
 		if err != nil {
 			progress.Fail() // progress.Done is handled inline
 		}
 	}()
 
-	var (
-		version *fastly.Version
-	)
+	name, source := c.manifest.Name()
+	path, err := pkgPath(c.Path, progress, name, source)
+	if err != nil {
+		return err
+	}
 
-	// If path flag was empty, default to package tar inside pkg directory
-	// and get filename from the manifest.
-	if c.Path == "" {
-		progress.Step("Reading package manifest...")
+	undoStack := common.NewUndoStack()
+	defer func() { undoStack.RunIfError(out, err) }()
 
-		name, source := c.manifest.Name()
-		if source == manifest.SourceUndefined {
-			return fmt.Errorf("error reading package manifest")
+	serviceID, source := c.manifest.ServiceID()
+	if source != manifest.SourceUndefined {
```

Comment:
7. If the `fastly.toml` manifest already has a Service ID set, then we'll go ahead and clone the latest version.